docs/smart/principals.md

```diff
@@ -13,47 +13,60 @@ _Principals_ are a Clarity native type that represents a spending entity. This s
-
-The following user-defined function transfers an asset, in this case, tokens, between two principals:
-
-```
-(define (transfer! (sender principal) (recipient principal) (amount int))
-  (if (and
-        (not (eq? sender recipient))
-        (debit-balance! sender amount)
-        (credit-balance! recipient amount))
-    true
-    false))
+```cl
+(define-public (transfer-to-recipient! (recipient principal) (amount uint))
+  (if (check-condition)
+      (stx-transfer? amount tx-sender recipient)
+      (err u400)))
 ```
 
-The principal's signature is not checked by the smart contract, but by the virtual machine.
-
-
 ## Smart contracts as principals
 
-Smart contracts themselves are principals and are represented by the smart contract's identifier. You create the identifier when you launch the contract, for example, the contract identifier here is `hanomine`.
+Smart contracts themselves are principals and are represented by the
+smart contract's identifier -- which is the publishing address of the
+contract _and_ the contract's name, e.g.:
 
-```bash
-clarity-cli launch hanomine /data/hano.clar /data/db
+```cl
+'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR.contract-name
 ```
 
-A smart contract may use the special variable `contract-name` to refer to its own principal.
+For convenience, smart contracts may write a contract's identifier in the
+form `.contract-name`, which is expanded by the Clarity interpreter into
+a fully-qualified contract identifier that corresponds to the same
+publishing address as the contract it appears in.
 
-To allow smart contracts to operate on assets it owns, smart contracts may use the special `(as-contract expr)` function. This function executes the expression (passed as an argument) with the `tx-sender` set to the contract's principal, rather than the current sender. The `as-contract` function returns the value of the provided expression.
+In order for a smart contract to operate on assets it owns, smart contracts
+may use the special `(as-contract ...)` function. This function
+executes the expression (passed as an argument) with the `tx-sender`
+set to the contract's principal, rather than the current sender. The
+`as-contract` function returns the value of the provided expression.
 
-For example, a smart contract that implements something like a "token faucet" could be implemented as so:
+For example, a smart contract that implements something like a "token
+faucet" could be implemented as so:
 
 ```cl
 (define-public (claim-from-faucet)
-  (if (is-none? (fetch-entry claimed-before (tuple (sender tx-sender))))
+  (if (is-none? (fetch-entry claimed-before {sender: tx-sender}))
       (let ((requester tx-sender)) ;; set a local variable requester = tx-sender
         (begin
             (insert-entry! claimed-before {sender: requester} {claimed: true})
-            (as-contract (stacks-transfer! requester 1)))))
-      (err 1))
+            (as-contract (stacks-transfer? u1 tx-sender requester))))
```

Comment:
Is it `stacks-transfer?` or `stx-transfer?`

Reply:
Fixed in 1a0d04d